# Семинар 6 ДЗ

## Задание $1$
Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным $16$.

Найти доверительный интервал для оценки математического ожидания $\alpha$ с надежностью $0.95$, если выборочная средняя $M = 80$, а объем выборки $n = 256$.


## Решение

Для решения этой задачи используем формулу:

$P(\overline X + Z_{\frac{α}{2}} \cdot \dfrac{\sigma}{\sqrt{n}} \leq \mu \leq \overline X + Z_{1- \frac{α}{2}} \cdot \dfrac{\sigma}{\sqrt{n}}) = p$

Для расчетов используем python

In [1]:
from scipy import stats
import numpy as np

In [2]:
sigma = 16
p = 0.95
mu = 80
n = 256
alhpa = 1 - p

In [3]:
t1 = stats.norm.ppf(alhpa/2)
t2 = stats.norm.ppf(1 - alhpa/2)
t1, t2

(-1.959963984540054, 1.959963984540054)

In [4]:
left = mu + t1 * sigma / np.sqrt(n)
right = mu + t2 * sigma / np.sqrt(n)
left, right

(78.04003601545995, 81.95996398454005)

$P(78.04 \leq \mu \leq 81.96) = 0.95$

## Задание $2$ 
В результате $10$ независимых измерений некоторой величины $X$, выполненных с одинаковой точностью, получены опытные данные:

$6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1$

Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины $X$ при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью $0,95$.


## Решение

Для решения этой задачи используем формулу:

$P(\overline X + t_{\frac{α}{2}} \cdot \dfrac{\sigma}{\sqrt{n}} \leq \mu \leq \overline X + t_{1- \frac{α}{2}} \cdot \dfrac{\sigma}{\sqrt{n}}) = p$

Для расчетов используем python

In [5]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [6]:
mu = x.mean()
sigma = x.std(ddof=1)
n = len(x)

p = 0.95
alhpa = 1-p

In [7]:
t1 = stats.t.ppf(alhpa/2, df= n - 1)
t2 = stats.t.ppf(1 - alhpa/2, df= n - 1)
t1, t2

(-2.2621571627409915, 2.2621571627409915)

In [8]:
left = mu + t1 * sigma / np.sqrt(n)
right = mu + t2 * sigma / np.sqrt(n)
left, right

(6.267515851415713, 6.912484148584288)

$P(6.2675 \leq \mu \leq 6.9125) = 0.95$

## Задание $3$
Рост дочерей $175, 167, 154, 174, 178, 148, 160, 167, 169, 170$

Рост матерей  $178, 165, 165, 173, 168, 155, 160, 164, 178, 175$

Используя эти данные построить $95\%$ доверительный интервал для разности среднего роста родителей и детей.

## Решение

Формула для нахождения доверительного интервала для разности средних арифметических $µ_1-µ_2$ будет:

$Δ± t_{\frac{α}{2}} \cdot S_Δ$ , где $Δ= \overline {X_1} - \overline {X_2}$

$P(Δ + t_{\frac{α}{2}} \cdot S_Δ \leq µ_1-µ_2 \leq Δ + t_{1 - \frac{α}{2}} \cdot S_Δ) = p$

$S_Δ= \sqrt{\dfrac{D}{n_1} + \dfrac{D}{n_2} }$, где $n_1$ и  $n_2$ – объемы выборок.

$D=\dfrac{1}{2}\cdot(D_1+D_2)$, где   $D_1$ и $D_2$ – дисперсии обеих групп

Для расчетов используем python

In [9]:
x = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
y = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])

In [12]:
n1 = len(x)
n2 = len(y)

d1 = x.var(ddof=1)
d2 = y.var(ddof=1)

D = (d1 + d2)/2
S_delta = (d1/n1 + d2/n2)**0.5
d1, d2, D, S_delta

(91.06666666666666, 60.10000000000001, 75.58333333333334, 3.888015775002291)

In [11]:
p = 0.95
alhpa = 1 - p

t1 = stats.t.ppf(alhpa/2, df= 2*(n - 1))
t2 = stats.t.ppf(1 - alhpa/2, df= 2 * (n - 1))
t1, t2

(-2.10092204024096, 2.10092204024096)

In [13]:
left = x.mean() - y.mean() + t1 * S_delta
right = x.mean() - y.mean() + t2 * S_delta
left, right

(-10.068418034506857, 6.268418034506846)

$P(-10.0684 \leq µ_1-µ_2 \leq 6.2684) = 0.95$